rfcs/0185-redistribute-redistributable.md

```diff
@@ -0,0 +1,225 @@
+
+# Prior art
+[prior-art]: #prior-art
+
+According to [this discussion](https://github.com/NixOS/nixpkgs/issues/83433), the current status quo dates back to the 20.03 release meeting.
+More than four years have passed, and it is likely worth rekindling this discussion, especially now that we actually have a Steering Committee.
+
+Recent exchanges have been happening in [this issue](https://github.com/NixOS/nixpkgs/issues/83884).
+
+# Resolved questions
+
+### How large are the packages Hydra would need to additionally store?
+
+`nix-community`'s Hydra instance can give us approximations.
+Its `unfree-redist-full` channel is currently 215G large, including around 200G of NVidia kernel packages and 15G for all the rest of unfree redistributable software.
+Its `cuda` channel is currently 482G large.
+
+Currently, NixOS' hydra pushes around 2TB per month to S3, with rebuilds taken into account.
+Noteworthy is the fact that these 2TB are of compressed data.
+Hence, the expected increase would not be 700G per rebuild, but something lower than this, which is hard to pre-compute.
+
+Regardless, Hydra should be able to deal pretty well even with a one-time 700G data dump.
+The issues would come only if compression were not good, in addition to rebuilds being frequent enough to significantly increase the amount of data Hydra pushes to S3.
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+Is the list of installation methods correct?
+I took it from my personal history as well as the NixOS website, but there may be others.
+Also, I may have the wrong job name, as I tried to guess the correct job name from the various links.
+
+Do we need a specific `redistributableWhenPatched` field on the license?
+It feels like this would be a bit too much, and probably `redistributable` would be enough.
+However, we may need to have it still.
+
+Will we need to redistribute some derivations with `runnableOnHydra = false`?
+For example, some firmware might not be legal to run on hydra.
+However, Hydra will never actually try to run it, as it cannot be used at runtime to build other packages.
+Maybe even `runnableOnHydra` could be better named to encompass this case too?
+
+# Future work
+[future]: #future-work
+
+- **Actually tagging licenses and packages as `runnableOnHydra`.**
+  Without this, this RFC would have no impact.
+  This will be done package-by-package, and should require no RFC, unless there are significant disagreements on whether a license should be runnable on hydra or not.
+
+- **Monitoring Hydra to confirm it does not push too much data to S3.**
+  If this change causes Hydra to push an economically non-viable amount of data to S3, then we should revert the addition of `runnableOnHydra` to the relevant packages and reconsider.
+
+- **Culling NVidia kernels and CUDA derivations.**
+  We suggest not caring too much about S3 size increases in the first step, considering the numbers from the resolved questions section.
```

Comment:
(Again I think these are by far the scariest cache size growth rate numbers I’ve heard in a long time, unless I’m misunderstanding them.)

Reply:
(same here)